src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,266 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.OData.Json
+{
+    /// <summary>
+    /// Implement of <see cref="IJsonWriter"/> that is based on
+    /// <see cref="Utf8JsonWriter"/>.
+    /// </summary>
+    internal class ODataUtf8JsonWriter : IJsonWriter, IDisposable
+    {
+        private const int DefaultBufferSize = 16 * 1024;
+        private const float BufferFlushThreshold = 0.9f;
+        private readonly static byte[] Parentheses = new byte[] { (byte)'(', (byte)')' };

```

Comment:
Marking this `readonly` doesn't stop someone from writing `Parentheses[0] = 0`. Do we have a `IReadOnlyList` implementation that we could use here?

Reply:
Wanted to use `IReadOnlySpan<byte>`. Not sure why I didn't, but this is now obsolete since I can use `WriteByte()`